ERCS/erc-7710.md

```diff
@@ -0,0 +1,177 @@
+delegation mechanism to ERC-1271, we ensure that delegations can only be created by contracts capable of demonstrating
+ownership and control over their actions, thereby reducing the risk of malicious delegations.
+
+**Flexibility and Scalability**: The proposed interfaces are designed to be minimal yet powerful, allowing contracts to
+delegate a wide range of actions without imposing a heavy implementation burden. This balance aims to encourage
+widespread adoption and innovation.
+
+**Interoperability**: Compatibility with existing standards, such as ERC-1271 and ERC-4337, ensures that our proposal
+can be seamlessly integrated into the current Ethereum infrastructure. This encourages adoption and leverages existing
+security practices.
+
+**Usability**: By enabling contracts to delegate specific actions to others, we open the door to more user-friendly
+DApps that can perform a variety of tasks on behalf of users, reducing the need for constant user interaction and
+enhancing the overall user experience.
+
+This ERC represents a step towards a more interconnected and flexible Ethereum ecosystem, where smart contracts can more
+effectively collaborate and adapt to users' needs.
+
+### Execution Interface
+
+A previous iteration of this spec defined `Action` as a simple `(target, value, data)` tuple, and defined a specific
+execution interface on the delegator that is `executeDelegatedAction(Action _action)` which the Delegation Manager is
+supposed to call.
+
+That approach had a few downsides:
+
+- Existing smart accounts won't be compatible with this spec (unless they happen to implement the execution interface).
+- The execution behavior is limited to a single call, since `Action` could only encode a single call.  It made complex
+  execution behaviors such as batching, delegatecall, and CREATE2 impossible.
+
+To solve the first issue, we decided to remove the requirement for the delegator to implement any specific interface.
+Rather, we rely on the Delegation Manager to correctly call the delegator, and we rely on the fact that a delegator would
+only create `_authority` for a Delegation Manager that knows how to correctly call it.
+
+To solve the second issue, we decied to adopt the execution interface from ERC-7579, which had to solve a similar problem
+within the context of modular smart accounts: defining a standardized execution interface that can support many types of
+executions.
+
+## Security Considerations
+
+The introduction of customizable authorization terms requires careful consideration of how authorization data is
+structured and interpreted. Potential security risks include the misinterpretation of authorization terms and
+unauthorized actions being taken if the interface is not properly implemented. It is recommended that applications
+implementing this interface undergo thorough security audits to ensure that authorization terms are handled securely.
+
+Needs discussion.
```

Comment:
```suggestion
Needs discussion. <!-- TODO -->
```

If you use an HTML-style comment, the linter will make sure you remove it before moving to Review.

Reply:
comment added.